### PR TITLE
Fix some colocation bugs

### PR DESCRIPTION
**Pull request by cowfee**
Fix some bugs discovered while testing out the colocated architecture on Goerli staging. Here's the related [infra PR](https://github.com/cowprotocol/infrastructure/pull/478).

The plan was to skip the release next week due to the hackathon. Maybe we should reconsider that, and deploy these changes so we can test things more thoroughly.

### CODE REVIEW

**Comment by MartinquaXD** [comment]
> The plan was to skip the release next week due to the hackathon. Maybe we should reconsider that, and deploy these changes so we can test things more thoroughly.

I think we don't have to deploy to prod to continue testing. Wouldn't we already get good enough info when these changes also get applied to `mainnet` and `gc` on staging?

**Review by MartinquaXD — COMMENTED** [review]
> The plan was to skip the release next week due to the hackathon. Maybe we should reconsider that, and deploy these changes so we can test things more thoroughly.

I think we don't have to deploy to prod to continue testing. Wouldn't we already get good enough info when these changes also get applied to mainnet and gc on staging?

**Comment by cowfee** [comment]
> Wouldn't we already get good enough info when these changes also get applied to mainnet and gc on staging?

Ah, good point, we're only skipping the prod release!

**Comment by nlordell** [comment]
Yes, I think we should progress with the experiment on Gnosis Chain and Mainnet **staging**.

**Review by nlordell — APPROVED** [review]
Nice catch!

Yes, we can deploy co-location on Gnosis Chain and Mainnet staging.

File: crates/solvers/src/infra/config/baseline/file.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ pub async fn load(path: &Path) -> super::Config {
         .await
         .unwrap_or_else(|e| panic!("I/O error while reading {path:?}: {e:?}"));
     let config = toml::de::from_str::<Config>(&data)
-        .unwrap_or_else(|_| panic!("TOML syntax error while reading {path:?}"));
+        .unwrap_or_else(|e| panic!("TOML syntax error while reading {path:?}: {e:?}"));
```
[the remark this answers is not in the document]
Comment (nlordell) [reply crates/solvers/src/infra/config/baseline/file.rs:48]: Good catch @MartinquaXD

**Pull request merged**